### ISSUE TITLE
Don't require Microsoft Build Tools on 3.10+ Pythons

### ISSUE THREAD

**Issue by ptsavol** [spine-tools/Spine-Toolbox#2524]
I think the culprit here is the `psutil` package, which is a dependency of our dependency (not sure which one). It seems that we have locked the psutil version to < 5.9.2 in toolbox. 5.9.1 requires the Microsoft build tools but newer versions of psutil don't. I'll just remove the psutil requirement lock and see what happens.

**Comment by PekkaSavolainen** [comment]
It works! The problem is caused by missing wheels on PyPi for Python 3.10+ which resulted into this error when doing `pip install -r requirements.txt`:

```
...
error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
...
```

In our current requirements, there are two packages that do not have wheels on PyPi for Python 3.11: `psutil` and `cx_oracle`. Dealing with psutil was easy, because all I had to do was to remove the `psutil<5.9.2` limit from toolbox. But cx_oracle does not have wheels for Python 3.11 either. The reason is that the package has been renamed to `oracledb` in the new **9** version.

@soininen and @manuelma is it Ok if I update spinedb_api's requirements.txt to install `oracledb` instead of `cx_oracle`? We don't actually use cx_oracle anywhere so I'm not sure why that even is there but I guess this might affect some user scripts. Or should we just remove cx_oracle from our requirements?

**Comment by soininen** [comment]
> is it Ok if I update spinedb_api's requirements.txt to install oracledb instead of cx_oracle?

I think we can drop `cx_oracle`. If we do, `"oracle"` should also be dropped from `spinedb_api.helpers.UNSUPPORTED_DIALECTS`. The requirements was added to `spinedb_api` ~3 months ago. It was not needed before and I do not think it will be needed in the future either. I just added it for completeness sake.

The `cx_oracle` requirement raises from SQLAlchemy which needs it to access Oracle databases.

We do not (yet) support Oracle databases for Spine data (only SQLite and MySQL are supported). The only use for `cx_oracle` is if somebody wants to import/export data or create an URL to such DB in Data Connection.

**Comment by PekkaSavolainen** [comment]
This should be fixed now in 0.8-dev branches. Anybody want to try this out before I merge this to master as well?

I also updated readme.md so that visual studio build tools are not mentioned there anymore and we now suggest using Python 3.11 instead of Python 3.9.

**Comment by soininen** [comment]
Nice work! Hopefully the performance improvements of Python 3.11 will show up in Toolbox as well.

**Comment by soininen** [comment]
@PekkaSavolainen Can this issue be closed now?

**Comment by PekkaSavolainen** [comment]
I'm 95% sure this is fixed but I was waiting on somebody to confirm. I guess I should merge the fix to master as well and then close it.

**Comment by PekkaSavolainen** [comment]
This is still a problem. Tried pipx install on Win11 with Python 3.12.3 installed from the Microsoft Store and this happened.

```
C:\Users\ttepsa>python -m pipx install spinetoolbox
Fatal error from pip prevented installation. Full pip output in file:
    C:\Users\ttepsa\pipx\logs\cmd_2024-06-06_11.43.29_pip_errors.log

pip failed to build package:
    gdxcc

Some possibly relevant errors from pip install:
    error: subprocess-exited-with-error
    error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/

Error installing spinetoolbox.
```

Looks like there are no `gdxcc` wheels on PyPI for Python 3.11 nor 3.12. Python 3.10 wheels are the latest ones.

**Comment by ptsavol** [comment]
This should now be fixed for Python 3.12!

I'll expect this same thing to happen when Python 3.13 is released (at least in the beginning of it's life cycle).
